crates/datis-core/src/extract.rs

```diff
@@ -51,58 +51,120 @@ pub fn extract_atis_station_frequencies(situation: &str) -> HashMap<String, Stat
 }
 
 pub fn extract_atis_station_config(config: &str) -> Option<StationConfig> {
-    let re = RegexBuilder::new(
-        r"^ATIS ([a-zA-Z- ]+) ([1-3]\d{2}(\.\d{1,3})?)(,[ ]?TRAFFIC ([1-3]\d{2}(\.\d{1,3})?))?(,[ ]?VOICE ([a-zA-Z-:]+))?(,[ ]?INFO ([a-zA-Z]))?$",
-    )
-    .case_insensitive(true)
-    .build()
-    .unwrap();
-
-    re.captures(config).map(|caps| {
-        let name = caps.get(1).unwrap().as_str();
-        let atis_freq = caps.get(2).unwrap().as_str();
-        let atis_freq = (f64::from_str(atis_freq).unwrap() * 1_000_000.0) as u64;
-        let traffic_freq = caps
-            .get(5)
-            .map(|freq| (f64::from_str(freq.as_str()).unwrap() * 1_000_000.0) as u64);
-        let tts = caps
-            .get(8)
-            .and_then(|s| TextToSpeechProvider::from_str(s.as_str()).ok());
-        let info_ltr = caps
-            .get(10)
-            .map(|ilo| ((ilo.as_str()).chars().next().unwrap().to_ascii_uppercase()) as char);
-        StationConfig {
-            name: name.to_string(),
-            atis: atis_freq,
-            traffic: traffic_freq,
-            tts,
-            info_ltr_override: info_ltr,
+    let re = RegexBuilder::new(r"^ATIS ([a-zA-Z- ]+) ([1-3]\d{2}(\.\d{1,3})?)")
+        .case_insensitive(true)
+        .build()
+        .unwrap();
+
+    let caps = re.captures(config).unwrap();
+    let name = caps.get(1).unwrap().as_str().to_string();
+    let atis_freq = caps.get(2).unwrap();
+    let atis_freq = (f64::from_str(atis_freq.as_str()).unwrap() * 1_000_000.0) as u64;
+
+    let mut traffic_freq: Option<u64> = None;
+    let mut tts: Option<TextToSpeechProvider> = None;
+    let mut info_ltr_override = None;
+
+    let rex_option = RegexBuilder::new(r"([^ ]*) (.*)")
+        .case_insensitive(true)
+        .build()
+        .unwrap();
+    for token in config.split(",").skip(1) {
+        let caps = rex_option.captures(token.trim()).unwrap();
+        let option_key = caps.get(1).unwrap().as_str();
+        let option_value = caps.get(2).map_or("", |m| m.as_str());
```

Comment:
Might not be necessary to use regex here

```suggestion
        let token = token.tim();
        let (key, value) = token.split_at(token.find(',').unwrap_or(token.len()));
```

(code is untested)

Reply:
Good one.  Done.